pkg/kubectl-nfd/validate.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectlnfd
+
+import (
+	"fmt"
+	"os"
+
+	"sigs.k8s.io/yaml"
+
+	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/v1alpha1"
+	"sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/validate"
+)
+
+// Given a file path, read the file and check if is a valid NodeFeatureRule file
+func ValidateNFR(filepath string) []error {
+	var err error
+	var validationErr []error
+
+	file, err := os.ReadFile(filepath)
+	if err != nil {
+		return []error{fmt.Errorf("error reading NodeFeatureRule file: %v", err)}
+	}
+
+	nfr := nfdv1alpha1.NodeFeatureRule{}
+	err = yaml.Unmarshal(file, &nfr)
+	if err != nil {
+		return []error{fmt.Errorf("error reading NodeFeatureRule file: %v", err)}
+	}
+
+	for _, rule := range nfr.Spec.Rules {
+		fmt.Println("Validating rule: ", rule.Name)
+		// Validate Rule Name
+		if rule.Name == "" {
+			validationErr = append(validationErr, fmt.Errorf("rule name cannot be empty"))
+		}
+
+		// Validate Annotations
+		validationErr = append(validationErr, validate.Annotations(rule.Annotations)...)
+
+		// Validate labels
+		validationErr = append(validationErr, validate.Labels(rule.Labels)...)
```

Comment:
I think we should handle the dynamic values of labels and ERs here. If value is in the format `@<part-1>.<part-2>` (or just starts with `@` to make it simple) just replace with a dummy value "0" or smth